### ISSUE TITLE
rule generator: insert default child statement when specifying basic block subscope

### ISSUE THREAD

**Issue by Ana06** [fireeye/capa#472]
The rule generator recognizes the following as a valid rule (Rule compiled and matched):
```
  features:
    - and:
      - offset: 0x10
      - offset: 0x14
```

But it complains that when adding a basic block (failed to match rule):
```
  features:
    - and:
      - offset: 0x10
      - basic block:
        - offset: 0x14
```

Adding an extra element to the basic block, it claims that it failed to compile the rule (invalid rule: subscope must have exactly one child statement):
```
  features:
    - and:
      - basic block:
        - offset: 0x14
        - offset: 0x10
```

https://user-images.githubusercontent.com/16052290/110840578-2e201000-82a5-11eb-99e5-98c57fea8199.mov

It looks like something is wrong with basic blocks as both should be valid rules and match.

**Comment by mr-tz** [comment]
Those are not valid rules, see line 351-352. Otherwise, how should the features be combined?

https://github.com/fireeye/capa/blob/14e65c46011aca908863a770ec925cc222dc7f93/capa/rules.py#L347-L354

To make this easier the explorer could automatically add an `and`/`or` under the basic block.

**Comment by mike-hunhoff** [comment]
Thank you for highlighting this @Ana06. @mr-tz is correct, these are not valid rules as the `basic block` subscope requires at least one child statement.

To make this less confusing we can insert a child statement when a user changes to `basic block` making sure to remove this child statement if the user switches to a different statement e.g. `and`, `or`.

**Comment by Ana06** [comment]
@mr-tz 
> Those are not valid rules

Although the second rule error is `failed to match rule` and not `invalid rule` 🤔

**Comment by mike-hunhoff** [comment]
Hm I'm guessing the second rule is "valid" in the sense that it contains exactly one child element, but capa won't be able to find a match for it because the type of the child statement is not as expected. If you nested `- offset: 0x14` underneath an `and` or `or` statement then the rule is valid and capa should be able to find a match for it.

We should modify the check to verify that a `subscope` has exactly one child element AND it's of the expected type.

Thoughts @mr-tz?

**Comment by mr-tz** [comment]
Yes, I think checking the type makes sense.

**Comment by mike-hunhoff** [comment]
addressed in #493.
